README.md

```diff
@@ -184,11 +184,21 @@ CMSIS package will generate linker script for your device automatically (target
 
 * `stm32_get_chip_info(<chip> [FAMILY <family>] [TYPE <type>] [DEVICE <device>])` - classify device using name, will return device family (into `<family>` variable), type (`<type>`) and canonical name (`<device>`, uppercase without any package codes)
 * `stm32_get_memory_info((CHIP <chip>)|(DEVICE <device> TYPE <type>) [FLASH|RAM|CCRAM|STACK|HEAP] [SIZE <size>] [ORIGIN <origin>])` - get information about device memories (into `<size>` and `<origin>`). Linker script generator uses values from this function
-* `stm32_get_devices_by_family(DEVICES [FAMILY <family>])` - return into `DEVICES` all supported devices by family (or all devices if `<family>` is empty)
 * `stm32_print_size_of_target(<target>)` - Print the application sizes for all formats
 * `stm32_generate_binary_file(<target>)` - Generate the binary file for the given target
 * `stm32_generate_hex_file(<target>)` - Generate the hex file for the given target
 
+To use the following functions, include the `devices.cmake` file with the following line
+
+```cmake
+include(${STM32_CMAKE_PATH}/cmake/stm32/devices.cmake)
+```
+
+where `STM32_CMAKE_PATH` is the path to the `stm32-cmake` repository.
+
+* `stm32_get_devices_by_family(DEVICES [FAMILY <family>])` - return into `DEVICES` all supported devices by family (or all devices if `<family>` is empty)
+* `stm32_print_devices_by_family([FAMILY <family>])` - Print all supported devices by family (or all devices if `<family>` is empty)
```

Comment:
For me this makes more sense.
```suggestion
* `stm32_get_devices_by_family(DEVICES <family>)` - return into `DEVICES` all supported devices by family (or all devices if `<family>` is empty)
* `stm32_print_devices_by_family(<family>)` - Print all supported devices by family (or all devices if `<family>` is empty)
```
`FAMILY` is only needed when you have multiple possible keywords.

Reply:
Hmm I thought `FAMILY` has to be specified as a keyword. Does it work without it?

Reply:
No I mean you could change the implementation so that FAMILY is not used. 
It brings nothing here.

Reply:
I think it is okay that way. I think this was done bcause https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html is used. Using it also allows easier extensibility (e.g. you still want to parse multiple families, but you also want to pass an option or something like a filter)

Reply:
> Using it also allows easier extensibility (e.g. you still want to parse multiple families, but you also want to pass an option or something like a filter)

OK I understand the point for evolution even if I don't think this function will evolve any soon.
cmake_parse_arguments is fine. I didn't know it before but it permits to do clean things.

I think if you want to have a clean function then all misuse errors should be managed
```cmake
stm32_print_devices_by_family(H7) # this shall throw error because of no FAMILY given
stm32_print_devices_by_family(FAMILY H7 L4) #this should throw error OR change FAMILY to a "multi_value_keywords"
stm32_print_devices_by_family(FAMILYH7) # this should throw error unknown keyword or something
stm32_print_devices_by_family(FAMILY) # this should throw error missing argument
```